compiler/src/codegen/compcore.re

```diff
@@ -277,80 +220,12 @@ let grain_function_imports = [
     mimp_setup: MSetupNone,
     mimp_used: false,
   },
-  {
-    mimp_mod: data_structures_mod,
-    mimp_name: new_rational_ident,
-    mimp_type:
-      MFuncImport(
-        [
-          Types.StackAllocated(WasmI32),
-          Types.StackAllocated(WasmI32),
-          Types.StackAllocated(WasmI32),
-        ],
-        [Types.StackAllocated(WasmI32)],
-      ),
-    mimp_kind: MImportWasm,
-    mimp_setup: MSetupNone,
-    mimp_used: false,
-  },
-  {
-    mimp_mod: data_structures_mod,
-    mimp_name: new_float32_ident,
-    mimp_type:
-      MFuncImport(
-        [Types.StackAllocated(WasmI32), Types.StackAllocated(WasmF32)],
-        [Types.StackAllocated(WasmI32)],
-      ),
-    mimp_kind: MImportWasm,
-    mimp_setup: MSetupNone,
-    mimp_used: false,
-  },
-  {
-    mimp_mod: data_structures_mod,
-    mimp_name: new_float64_ident,
-    mimp_type:
-      MFuncImport(
-        [Types.StackAllocated(WasmI32), Types.StackAllocated(WasmF64)],
-        [Types.StackAllocated(WasmI32)],
-      ),
-    mimp_kind: MImportWasm,
-    mimp_setup: MSetupNone,
-    mimp_used: false,
-  },
-  {
-    mimp_mod: data_structures_mod,
-    mimp_name: new_int32_ident,
-    mimp_type:
-      MFuncImport(
-        [Types.StackAllocated(WasmI32), Types.StackAllocated(WasmI32)],
-        [Types.StackAllocated(WasmI32)],
-      ),
-    mimp_kind: MImportWasm,
-    mimp_setup: MSetupNone,
-    mimp_used: false,
-  },
-  {
-    mimp_mod: data_structures_mod,
-    mimp_name: new_int64_ident,
-    mimp_type:
-      MFuncImport(
-        [Types.StackAllocated(WasmI32), Types.StackAllocated(WasmI64)],
-        [Types.StackAllocated(WasmI32)],
-      ),
-    mimp_kind: MImportWasm,
-    mimp_setup: MSetupNone,
-    mimp_used: false,
-  },
   {
     mimp_mod: equal_mod,
     mimp_name: equal_ident,
     mimp_type:
       MFuncImport(
-        [
-          Types.StackAllocated(WasmI32),
-          Types.StackAllocated(WasmI32),
-          Types.StackAllocated(WasmI32),
-        ],
+        [Types.HeapAllocated, Types.HeapAllocated, Types.HeapAllocated],
```

Comment:
Why is this changed to heap allocated?

Reply:
Was just wrong before. This is a little bit of a subtle bug to catch too, since it'd only be a problem if you were pattern matching on heap-allocated constants. I only noticed it because of the refactor.